src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -592,7 +599,29 @@ private void startAgent(SlaveComputer computer, final TaskListener listener, Str
                             String workingDirectory) throws IOException {
         session = connection.openSession();
         expandChannelBufferSize(session,listener);
-        String cmd = "cd \"" + workingDirectory + "\" && " + java + " " + getJvmOptions() + " -jar " + AGENT_JAR +
+
+        PrintStream logger = listener.getLogger();
+
+         // Code to support alternate drives (such as D:\) for the remote root directory on Windows, when not using cygwin.
+         // You can disable this feature with startup option: -Dhudson.plugins.sshslaves.alternativewindowsdrive=false
+        boolean alternativeWindowsDriveCheck = ! "false".equals(System.getProperty("hudson.plugins.sshslaves.alternativewindowsdrive"));
+        String agentOSTYPE="";
+        try {
+            ByteArrayOutputStream agentOsTypeStream = new ByteArrayOutputStream();
+            connection.exec("echo \"$OSTYPE\"", agentOsTypeStream);
+            agentOSTYPE = StringUtils.chomp( agentOsTypeStream.toString(Charset.defaultCharset().name()) );
+        } catch (InterruptedException ex) {
+            // exceptions here will only disable the alternativeWindowsDriveCheck
+            logger.println(Messages.SSHLauncher_UnableToGetEnvironment(getTimestamp()));
+            alternativeWindowsDriveCheck=false;
```

Comment:
```suggestion
            alternativeWindowsDriveCheck=false;
            Thread.currentThread().interrupt();
```

Catching an interrupt and not resetting the interrupt flag will lead to zombie threads or some other bad behavior, I forget which.